### DISCUSSION TITLE
API programmatic : upload binary file like favicon.ico

### DISCUSSION THREAD

**Discussion by Geminii** [codesandbox/codesandbox-client#4098] [General] [answered]
Hi codesandbox team and community :) 

I'm trying to upload programmatically a local project into codesandbox. 
I follow the documentation about it and trying this code : 
```
{
  "files": {
    "static/favicon.ico": {
      "content": fs.readFileSync(file, 'binary') // i try Buffer.from(data, 'binary').toString('base64')
      "isBinary": true
    }
  }
}
```

Did i miss something about documentation. There is poor information about it :( 
Thanks a lot for feedbacks :heart::heart:

**Comment by garethx** [comment, marked as answer]
No, only the CLI supports binary file creation like this right now. What's your use-case?

**Reply by Geminii** [reply]
hi @garethx 
Thanks for the answer. Ok i understand why it was impossible to upload my favicon into codesandbox. 
It was at goal to initialize a default project with some basic static like favicon.ico or another picture. 
In the documentation, what does that mean *isBinary* property so ?

**Reply by garethx** [reply]
Ah, I will document this but if `isBinary` is `true`, then content needs to be set to a URL. So in your case the URL could specify an external favicon to be used.

e.g.
```
{
  isBinary: true,
  content: 'https://...'
}
```

**Reply by Geminii** [reply]
Ah ok. Don’t understand this in fact. 
But i try to upload the favicon as base64 with isBinary parameter 😮 nice try lol 
Thanks a lot for precision about it 😀
